File: json-mapper/src/main/java/com/anjunar/json/mapper/annotations/UseConverter.java
package com.anjunar.json.mapper.annotations;

import com.anjunar.json.mapper.converter.JacksonJsonConverter;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.FIELD, ElementType.METHOD})
public @interface UseConverter {

    Class<? extends JacksonJsonConverter> value() default JacksonJsonConverter.class;

}

File: json-mapper/src/main/kotlin/com/anjunar/json/mapper/annotations/UseConverter.kt
package com.anjunar.json.mapper.annotations

import com.anjunar.json.mapper.converter.Converter
import com.anjunar.json.mapper.converter.JacksonJsonConverter
import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.PROPERTY)
annotation class UseConverter(val value : KClass<out Converter> = JacksonJsonConverter::class)
